manager/src/main/java/io/gingersnapproject/Caches.java

```diff
@@ -60,6 +65,26 @@ private void actualRemove(String name, String key, Uni<String> value) {
       }
    }
 
+   public  Uni<String> denormalizedPut(String name, String key, String value) {
+      Table table = databaseHandler.table(configuration.rules().get(name).connector().table());
+      if (!table.foreignKeys().isEmpty()) {
+         // We have foreign keys to resolve
+         Json json = Json.read(value);
+         UniJoin.Builder<Object> builder = Uni.join().builder();
+         for (ForeignKey fk : table.foreignKeys()) {
+            // TODO: handle composite fks
+            Json fkJson = json.atDel(fk.columns().get(0));
+            String fkId = fkJson.asString();
+            String fkRule = databaseHandler.tableToRuleName(fk.refTable());
+            builder.add(get(fkRule, fkId).map(Json::read).map(j -> json.set(fkRule, j)));
```

Comment:
This can count as a hit/miss in the metrics, is it an issue?

Reply:
That is a good question about what to do for stats with the foreign keys... I am not sure the answer myself either.

Another very weird edge case is since these gets are not transactional, it is possible to find a foreign key but when we query the database the row doesn't exist. I think we may need to possibly think about doing a join to populate all the expanded foreign key columns as well.

Reply:
I have replaced the call to `get(fkRule,fkId)` with `getOrCreateMap(fkRule).get(fkId)` which will avoid metric pollution